### PR TITLE
feat(seekable-ranges): Add media-seekable/media.seekable state suppor…

### PR DESCRIPTION
**Pull request by cjpillsbury**
…t. Update media-time-display and media-time-range to handle seekable.

- Add state management for seekable (start & end) to `<media-controller/>`
- Add initial support for seekable start/end to `<media-time-range/>` and `<media-time-display/>` (note: no support for *both* seekable.end and (expected) duration (yet!)
- Update range and related math to no longer assume input.min will always be 0.

relates to #218

### CODE REVIEW

File: src/js/media-controller.js [hunk 1]
```diff
@@ -682,9 +693,17 @@ const getCurrentTime = (controller) => {
 };
 
 const getDuration = (controller) => {
-  const media = controller.media;
+  const media = controller?.media;
+  if (!Number.isFinite(media?.duration)) return NaN;
```
Comment (luwes) [inline src/js/media-controller.js:697]: is this to handle `Infinity`?

File: src/js/media-time-range.js [hunk 2]
```diff
@@ -137,61 +145,58 @@ class MediaTimeRange extends MediaChromeRange {
...
+    const [start] = this.mediaSeekable ?? [];
+    return start;
   }
 
   /* Add a buffered progress bar */
   getBarColors() {
     let colorsArray = super.getBarColors();
+    const { range } = this;
+    // Use mediaCurrentTime instead of range.value for precision (CJP)
+    const value = this.mediaCurrentTime;
```
Comment (luwes) [inline src/js/media-time-range.js:186]: can we use `currentTime` instead of `value` for naming?

**Review by luwes — APPROVED** [review]
a small nit but LGTM

File: src/js/media-controller.js (hunk 1 above)
Comment (cjpillsbury) [reply src/js/media-controller.js:697]: yup, `Infinity` and any non-numerics all in one fell swoop.

**Pull request merged**